game-core/src/main/java/games/strategy/net/AbstractServerMessenger.java

```diff
@@ -157,9 +157,7 @@ private boolean isUsernameMutedInCache(final String username) {
   @Override
   public void notifyUsernameMutingOfPlayer(final String username, final @Nullable Instant muteExpires) {
```

Comment:
I'm confused, why is the Instant staying here?

Reply:
The *very* important perspective is that we are modifying the commands available to anyone hosting a game. We are also modifying the 'mini' bans/mutes which go away with the lifetime of JVM.

Giving such granularity on all options is a false feature. Just because it was easy to get a mac-ban table going, does not mean it was wise to copy paste that for user-names, IPs, and then to copy/paste that code again for 'mini-bans' which have a different context. We're seeing the code 6 times, combined with signs time was running short when it comes to the DB layer, this area is a good place to inspect for features that really work or really make sense. For example, take the confirmation text, it was not previewed. In code it looks okay, rendered, pretty clearly not. 

I do not mean any disrespect to this legacy, but knowing what it is, it adds context for why we have the current features and why the code looks the way it does.

Anyways, long story long, perma-mute does not really make sense, but for now leaving this alone. Mute perhaps does not make sense at all, perhaps either just kick or mini-ban. Ban durations in lobby is not being touched, having time limits there makes perfect sense. I do wonder a bit if 'perma' ban should have an upper time limit, something in the range of 1-3 months. That is a bit of  a side-note.

Perhaps that context makes more sense now, it's important to keep in mind this mostly an update for every-day game hosting players and the tools available to them

A permanent mute does not make sense (either that, or it's *really* passive aggressive). The key about 'ban' in this context is that it's the 'mini' ban and only lives for the lifetime of server or bot. For the first case, if someone is banned then chances are really good a 15 minute ban is not the desired outcome, and on the other side most servers are alive for hours to a fraction of a day. Hence any ban is effectively desired to be permanent. In bot case, it takes a moderator to intervene, if someone grabs a mod for something that is happening in a bot, and the mod is able to make the ban, then it's clear we want more than a multi-hour ban from that bot. Given bots are restarted after a few weeks, perhaps every other month, again, it's kinda giving false choice when the ban will go away, and 

The key is that a 'mini-ban' lives only for as long as the JVM. If banning a player from a game that is being played, there is no desire for it to be just 15 minutes, and the server is going to not last longer than a day anyways. For the bot case, if a moderator needs to get involved and we catch the player *in* the bot and get them banned, it's because they should be banned. Our bots are somewhat long lived, but not that long lived where the ban is actually permanent. 

Side-notes: remote-moderator mutes I don't think make sense at all, I'm not sure if 'mute' actually makes sense. But that is getting to another topic, when it comes to 'mini-bans', we do not need another copy/paste. The evolution is kinda evil, first there was mac bans, then it was copy pasted to username bans, then IP bans, then we created 'in-memory' "mini-bans" and copy/pasted all three to that... No disrespect to the history, it's pretty clear time was running tight when the lobby and database work was being done, the OG maintainers did eventually have jobs to run off to.